File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int t) {
        int[] ans = new int[2];
        HashMap<Integer, Integer> hm = new HashMap<>();

        for(int i = 0; i<nums.length; i++) {
            int x = nums[i];
            int y = t-x;
            if(hm.containsKey(y)) {
                ans = new int[]{i, hm.get(y)};
                return ans;
            } else hm.put(x, i);
        }

        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val hm = mutableMapOf<Int, Int>()
        val res = IntArray(2)
        
        for(i in 0 until nums.size) {
            val toFind = target - nums[i]

            if(hm.contains(toFind)) {
                res[0] = hm.get(toFind) ?: 0
                res[1] = i
                return res;
            }

            hm.put(nums[i], i)
        }

        return res
    }
}